lib/mix_unused/analyzers/unreachable.ex

```diff
@@ -22,11 +55,14 @@ defmodule MixUnused.Analyzers.Unreachable do
     reachables = graph |> Graph.reachable(usages) |> MapSet.new()
     called_at_compile_time = Calls.called_at_compile_time(data, exports)
 
+    root_checker = RootChecker.new(graph, config.root_only)
+
     for {mfa, _meta} = call <- exports,
         candidate?(call),
         mfa not in usages,
         mfa not in reachables,
         mfa not in called_at_compile_time,
+        RootChecker.isRoot?(root_checker, mfa),
```

Comment:
We can put the condition inline, the `RootChecker` module seems a bit overkill.
For instance:
```
not root_only or Graph.in_degree(graph, mfa) == 0
```

I would consider to make the inverse of `root_only`, like `report_transitively_unused`.